docs/book/src/topics/crd-changes/v1alpha7-to-v1alpha8.md

```diff
@@ -0,0 +1,48 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*
+
+- [v1alpha7 compared to v1alpha8](#v1alpha7-compared-to-v1alpha8)
+  - [Migration](#migration)
+  - [API Changes](#api-changes)
+    - [`OpenStackCluster`](#openstackcluster)
+      - [Change to externalNetworkID](#change-to-externalnetworkid)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# v1alpha7 compared to v1alpha8
+
+> ⚠️ v1alpha8 has not been released yet.
+
+## API Changes
+
+This only documents backwards incompatible changes. Fields that were added to v1alpha8 are not listed here.
+
+### `OpenStackCluster`
+
+#### Change to externalNetworkID
+
+The field `externalNetworkID` has been renamed to `externalNetwork` and isn't a string anymore but a reference to a `NetworkFilter` object.
+
+```yaml
+externalNetworkID: "..."
+```
+
+becomes
+
+```yaml
+externalNetwork:
+  id: "..."
+```
+
+It is now possible to specify a `NetworkFilter` object to select the external network to use for the cluster. The `NetworkFilter` object allows to select the network by name, by ID or by tags.
+
+```yaml
+externalNetwork:
+  name: "..."
+```
+
+If a network is provided, it'll be added to `OpenStackCluster.Status.ExternalNetwork`.
+If no network is provided, CAPO will try to find network marked "External" and add it to `OpenStackCluster.Status.ExternalNetwork`.
+If the network can't be found, `OpenStackCluster.Status.ExternalNetwork` will be set to nil.
```

Comment:
This implies that when network is provided but isn't found, the CAPO will silently ignore the error. I believe that's wrong and if I read the code correctly it currently has that behavior.

Reply:
yes exactly :(

Reply:
yeah I'm fine returning an error.

Reply:
Do you know why we strictly need an external network? If a cluster is deployed on a provider network and reachable from outside already, this should not be needed, right? (ie no needs of floating ips, etc).

Reply:
I'm pretty sure we have it for two reasons:

1. If we're creating a router, we need to point it to an external network to allow connectivity.
2. We need a network to create FIPs in (and later on to pass it to CCM so it creates FIPs there).

My main point here is that currently the code silently ignores when user specified a non-existing network. IMO it can just go over it when user haven't specified it, but this isn't really a good user experience if we don't inform users that their input was wrong.

Reply:
But:

1. Router creation is optional, right?
2. Why do we need FIPs? What about when deploying the cluster on a pre-existing network that is already routed (e.g. provider network)

Reply:
Ok we agreed offline that:
* external network is not strictly required
* if the user provides an external network we will change the behaviour to return a failure if that network wasn't found
* investigate how a user can specify that no external network is desired